lib/service-data/service-data.js

```diff
@@ -299,10 +299,13 @@ external.getInstanceIdByPropertyValue = (property, value) => {
 
 external.getInstanceTitle = (_id) => {
```

Comment:
I think we need a `getSummaryInstanceTitle`

Now if a summary version of a legend, label or text is set that's the value we'll end up seeing in the editor

![GOV_UK_-_The_best_place_to_find_government_services_and_information](https://user-images.githubusercontent.com/4856/57083844-1c58a380-6cf1-11e9-8130-0bf9cd02140d.png)

just riffing, thoroughly untested, but somethiing like this

```
external.getInstanceTitle = (_id, isSummary=false) => {
  const instance = external.getInstance(_id)

  if (!instance) {
    return
  }

  let title

  if (isSummary) {
    title = instance.summaryLegend || instance.summaryLabel || instance.summaryText
  }
  if (!title) {
    title = instance.title || instance.heading || instance.legend || instance.label || instance.summaryText || instance.text
  }

  if (!title && instance._type === 'page.singlequestion') {
    if (instance.components && instance.components[0]) {
      const firstComponent = instance.components[0]
      title = external.getInstanceTitle(firstComponent._id, isSummary)
    }
  }
  return title || _id
}


external.getSummaryInstanceTitle = (_id) => {
  return external.getInstance(_id, true)
}
```

and then obviously use the `getSummaryInstanceTitle` in the summary controller and wherever necessary

Reply:
And yes, we do need to cover the summary heading situation. I meant to mention that too but got carried away. (Updated the code snippet accordingly)

Really not sure we need the `instance.title` any more either

Reply:
`s/getSummaryInstanceTitle/getInstanceSummaryTitle/`

Reply:
Resolving this